llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -1546,6 +1540,34 @@ void VPSlotTracker::assignNames(const VPBasicBlock *VPBB) {
       assignName(Def);
 }
 
+std::string VPSlotTracker::getName(const VPValue *V) {
+  auto *UV = V->getUnderlyingValue();
+  auto *VPI = dyn_cast_or_null<VPInstruction>(V->getDefiningRecipe());
```

Comment:
Looks like the result of the dyn_cast is unconditionally dereferenced. It is not immediately clear why all VPValues w/o underlying value should be VPInstructions. Is there some check before calling getName that handles all other cases? If so, this should be `cast`, sunk into the condition